src/persist-client/src/internal/trace.rs

```diff
@@ -1113,6 +1137,27 @@ impl<T: Timestamp + Lattice> Spine<T> {
         }
     }
 
+    /// Apply some amount of effort to trace maintenance.
+    ///
+    /// The units of effort are updates, and the method should be thought of as
+    /// analogous to inserting as many empty updates, where the trace is
+    /// permitted to perform proportionate work.
+    ///
+    /// Returns true if this did work and false if it left the spine unchanged.
```

Comment:
I had thought that the caller could figure this out by whether the value of `effort` had changed, but it turns out that it is never modified in practice, either in our impl or upstream. I'd be tempted to remove the mut here to avoid confusion, but I'm open to the usual argument that it's nice to match upstream even when upstream is being surprising...